File: quickSort/quickSort.py
# Based on divide and conquer algorithm
# Best case scenario: O(nlogn)
# Worst case scenario: O(n^2)

def quicksort(array)
  if len(array) < 2:
    return array
  else:
    pivot = array[0]
    less = [i for i in array[1:] if i <= pivot]
    greater = [i for array[1:] if i > pivot]

    return quicksort(less) + [pivot] + quicksort(greater)

File: quickSort/quickSort.js
// Same algorithm in JS

function quicksort(array){
  if(array.length < 2) return array

  let pivot = array[0]
  let greater = []
  let lesser = []

  for (let i =1; i<array.length; i++){
    (array[i] < pivot) ? lesser.push(array[i]) : greater.push(array[i]);
  }
  // recursive case
  // left + pivot + right
  return [...quicksort(lesser),pivot,...quicksort(greater)]
}

var arr1 = [7,8,2,0,5,2]
console.log(`Unsorted array ${arr1}`)
console.log(`Sorted array ${quicksort(arr1)}`)
